match/lib/match/storage/s3_storage.rb

```diff
@@ -152,6 +159,17 @@ def generate_matchfile_content(template: nil)
 
       private
 
+      def s3_object_path(file_name)
+        santized = sanitize_file_name(file_name)
+        return santized if santized.start_with?(s3_object_prefix)
+
+        s3_object_prefix + santized
+      end
+
+      def strip_s3_object_prefix(object_path)
+        object_path.gsub(s3_object_prefix, "")
```

Comment:
This should be anchored to the front of the object_path just in case the s3_object_prefix is found multiple times in the path. Want to prevent this behavior
```
prefix = hello
object_path = 'hello_there_hello'
stripped = '_there_'
```

```suggestion
        object_path.gsub(/^#{s3_object_prefix}/, "")
```